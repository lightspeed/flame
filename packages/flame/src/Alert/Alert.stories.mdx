import { Story, Preview, Meta, Props } from '@storybook/addon-docs/blocks';
import { Alert } from './.';
import { DocsContainer } from '../../.storybook/preview';

<Meta
  title="Components/Alert"
  component={Alert}
  argTypes={{
    onClose: {
      type: null,
    },
    icon: {
      type: null,
    },
  }}
/>

# Alert

Use an alert when you want to signal that something is going on.

## Types

<Story name="types">
  <div>
    <h3>Info (default)</h3>
    <Alert title="Info Card" noCloseBtn>
      This is a Card
    </Alert>
    <h3>Danger</h3>
    <Alert title="Danger Card" type="danger" noCloseBtn>
      This is a Card
    </Alert>
    <h3>Success</h3>
    <Alert title="Success Card" type="success" noCloseBtn>
      This is a Card
    </Alert>
    <h3>Warning</h3>
    <Alert title="Warning Card" type="warning" noCloseBtn>
      This is a Card
    </Alert>
  </div>
</Story>

## Playground

<Preview>
  <Story name="basic">
    {props => (
      <Alert {...props} title={props.title || 'Title'}>
        This is an Alert component of type "{props.type}"
      </Alert>
    )}
  </Story>
</Preview>

## Props

<Props story="basic" />
