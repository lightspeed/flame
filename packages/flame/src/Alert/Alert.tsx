import * as React from 'react';
import styled from '@emotion/styled';
import { transparentize } from 'polished';
import { space, variant, SpaceProps } from 'styled-system';
import { themeGet } from '@styled-system/theme-get';
import { Flex, Box } from '../Core';
import { Text } from '../Text';

const alertStyles = variant({
  key: 'alertVariants',
  prop: 'type',
});

const AlertWrapper = styled('div')<{ type: string }>`
  display: flex;
  justify-content: space-between;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0, 0, 0, 0.06), 0 3px 6px 0 rgba(0, 0, 0, 0.03),
    0 1px 2px 0 rgba(0, 0, 0, 0.1);
  border-top: 4px solid;
  border-radius: ${themeGet('radii.radius-2')};
  padding: ${themeGet('space.2')} ${themeGet('space.3')};
  ${space}
  ${alertStyles}
`;

AlertWrapper.defaultProps = {
  type: 'info',
};

const CloseButton = styled('button')`
  font-size: ${themeGet('fontSizes.text')};
  color: ${themeGet('colors.textHeading')};
  background-color: ${props => transparentize(0.9, themeGet('colors.textHeading', '#000')(props))};
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;

  &:focus {
    outline: none;
  }
`;

CloseButton.displayName = 'CloseButton';

export interface AlertProps {
  /** CSS class name */
  className?: string;
  /** Enum for preset Alert types */
  type?: 'info' | 'success' | 'warning' | 'danger' | string;
  /** Function called when Close button is tapped */
  onClose?: Function;
  /** Whether a Close button appears */
  noCloseBtn?: boolean;
  /** Icon for the alert */
  icon?: React.ReactNode;
  /** Text for the alert's title */
  title?: string;
}

const Alert: React.FunctionComponent<AlertProps & SpaceProps> = ({
  children,
  type = 'info',
  title,
  noCloseBtn,
  onClose,
  icon = null,
  ...restProps
}) => {
  const [isHidden, setIsHidden] = React.useState(false);

  const handleClose = (e: React.MouseEvent<HTMLElement>) => {
    if (onClose && typeof onClose === 'function') onClose(e);
    setIsHidden(s => !s);
  };

  if (isHidden) return null;

  return (
    <AlertWrapper type={type} {...restProps}>
      <Flex flex="1" alignItems="flex-start">
        {icon && (
          <Flex alignItems="center" pr={2} pt="2px">
            {icon}
          </Flex>
        )}
        <Box flex="1">
          {title && (
            <Text color="textHeading" fontWeight="bold" fontSize="text" mt={0} mr={0} mb={1} ml={0}>
              {title}
            </Text>
          )}
          <Box fontSize="text-s">{children}</Box>
        </Box>
      </Flex>
      {!noCloseBtn && (
        <CloseButton type="button" onClick={handleClose}>
          <svg
            xmlns="http://www.w3.org/2000/svg"
            viewBox="0 0 8 8"
            css={{
              width: '0.5em',
              height: '0.5em',
              fill: 'black',
              position: 'absolute',
              top: '0.25em',
              left: '0.25em',
              right: '0',
              bottom: '0',
            }}
          >
            <g fillRule="evenodd" transform="translate(-4 -4)">
              <path
                fillOpacity=".5"
                d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
              />
            </g>
          </svg>
        </CloseButton>
      )}
    </AlertWrapper>
  );
};

export { Alert };
