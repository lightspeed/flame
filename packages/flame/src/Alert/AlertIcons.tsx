import * as React from 'react';

import { IconWarning } from '../Icon/Warning';
import { IconInfo } from '../Icon/Info';
import { IconDanger } from '../Icon/Danger';
import { IconVerified } from '../Icon/Verified';

interface AlertIcons {
  type: 'info' | 'success' | 'warning' | 'danger' | string;
}
const AlertIcons: React.FC<AlertIcons> = ({ type }) => {
  if (type === 'info') {
    return <IconInfo />;
  }

  if (type === 'success') {
    return <IconVerified />;
  }

  if (type === 'warning') {
    return <IconWarning />;
  }

  if (type === 'danger') {
    return <IconDanger />;
  }

  return null;
};

export { AlertIcons };
