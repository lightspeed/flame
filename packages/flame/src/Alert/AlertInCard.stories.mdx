import { Story, Preview, Meta, Props } from '@storybook/addon-docs/blocks';
import { AlertInCard } from './.';
import { Card, CardSection } from '../Card';
import { DocsContainer } from '../../.storybook/preview';

<Meta
  title="Components/AlertInCard"
  component={AlertInCard}
  argTypes={{
    onClose: {
      type: null,
    },
    type: {
      control: {
        type: 'select',
        options: ['info', 'warning', 'success', 'danger'],
      },
    },
  }}
/>

# AlertInCard

A specially stylized Alert meant to be used within the `<Card />` component.

## Types

<Story name="types">
  <div>
    <h3>Info (default)</h3>
    <AlertInCard noCloseBtn>This is an AlertInCard component</AlertInCard>
    <h3>Danger</h3>
    <AlertInCard type="danger" noCloseBtn>
      This is an AlertInCard component
    </AlertInCard>
    <h3>Success</h3>
    <AlertInCard type="success" noCloseBtn>
      This is an AlertInCard component
    </AlertInCard>
    <h3>Warning</h3>
    <AlertInCard type="warning" noCloseBtn>
      This is an AlertInCard component
    </AlertInCard>
  </div>
</Story>

## Playground

<Preview>
  <Story name="basic">
    {props => (
      <Card>
        <CardSection>
          <AlertInCard {...props}>
            This is an AlertInCard component of type "{props.type}"
          </AlertInCard>
        </CardSection>
      </Card>
    )}
  </Story>
</Preview>

## Props

<Props story="basic" />
