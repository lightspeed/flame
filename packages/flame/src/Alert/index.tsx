import { Alert, AlertProps } from './Alert';

export { Alert, AlertProps };
