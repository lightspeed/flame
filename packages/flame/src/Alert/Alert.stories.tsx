import * as React from 'react';

import { Alert } from './Alert';
import { AlertIcons } from './AlertIcons';
import { Text } from '../Text';
import { IconApple } from '../Icon/Apple';

export default {
  title: 'Components|Alert',
  component: Alert,
  includeStories: ['ToBeDeprecatedFeature', 'types'],
};

const alertTypes = ['info', 'warning', 'success', 'danger'] as const;

export const types = () => {
  return (
    <div>
      <h2>Alert Types</h2>
      {alertTypes.map(type => {
        return (
          <Alert
            key={type}
            type={type}
            title={`${type[0].toUpperCase().concat(type.slice(1))}`}
            mb={2}
          >
            Description of alert.
          </Alert>
        );
      })}
      <h2>Alert Types (no close button)</h2>
      {alertTypes.map(type => {
        return (
          <Alert
            key={type}
            type={type}
            title={`${type[0].toUpperCase().concat(type.slice(1))}`}
            icon={<AlertIcons type={type} />}
            mb={2}
            noCloseBtn
          >
            Description of alert with no close button.
          </Alert>
        );
      })}
    </div>
  );
};

types.story = {
  name: 'Alert',
};

export const ToBeDeprecatedFeature = () => {
  return (
    <div>
      <p>
        Custom icons will be deprecated. Instead, we will automatically assign icons (regardless of
        the presence of the icon prop) to an alert starting from next major version.
      </p>
      {alertTypes.map(type => {
        return (
          <Alert
            key={type}
            type={type as any}
            title={`${type[0].toUpperCase().concat(type.slice(1))}`}
            icon={<IconApple />}
            mb={2}
          >
            <Text as="p" m={0}>
              Description of alert.
            </Text>
          </Alert>
        );
      })}
    </div>
  );
};

ToBeDeprecatedFeature.story = {
  name: 'To be deprecated feature',
};
