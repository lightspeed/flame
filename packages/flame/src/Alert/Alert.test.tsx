import * as React from 'react';
import { createComponent, customRender, fireEvent } from 'test-utils';
import { Alert } from './Alert';

describe('Alert', () => {
  it('renders correctly', () => {
    const component = createComponent(<Alert>Hello World</Alert>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  ['success', 'info', 'warning', 'danger'].forEach(type => {
    it(`renders type ${type} correctly`, () => {
      const component = createComponent(
        <Alert key={type} type={type}>
          Hello World
        </Alert>,
      );
      expect(component.toJSON()).toMatchSnapshot();
    });
  });

  it('renders a number', () => {
    const component = createComponent(<Alert>{300}</Alert>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders with an icon', () => {
    const { getByTestId } = customRender(
      <Alert icon={<i data-testid="icon" />}>Hello World</Alert>,
    );
    expect(getByTestId('icon')).toBeInTheDocument();
  });

  it('renders with a title', () => {
    const component = createComponent(<Alert title="My Title">Hello World</Alert>);
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('Removes close button', () => {
    const component = createComponent(
      <Alert title="My Title" noCloseBtn>
        Hello World
      </Alert>,
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  it('accepts a custom handleClose function', () => {
    const handleClose = jest.fn();
    const { container } = customRender(<Alert onClose={handleClose}>Hello World</Alert>);

    expect(handleClose).not.toHaveBeenCalled();

    fireEvent.click(container.querySelector('button'));

    expect(handleClose).toHaveBeenCalledTimes(1);
  });
});
