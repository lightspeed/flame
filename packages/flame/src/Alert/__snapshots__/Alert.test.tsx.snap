// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Alert Removes close button 1`] = `
.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e9fafe;
  border-color: #2875c6;
}

.emotion-6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2 {
  font-size: 0.875rem;
}

.emotion-0 {
  margin: 0;
  color: #0c0d0d;
  font-weight: 700;
  font-size: 1rem;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0.375rem;
  margin-left: 0;
  font-weight: 700;
}

<div
  className="emotion-8 emotion-9"
  type="info"
>
  <div
    className="emotion-6 emotion-7"
  >
    <div
      className="emotion-4 emotion-3"
    >
      <p
        className="emotion-0 emotion-1"
        color="textHeading"
        fontSize="text"
        fontWeight="bold"
      >
        My Title
      </p>
      <div
        className="emotion-2 emotion-3"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
</div>
`;

exports[`Alert renders a number 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e9fafe;
  border-color: #2875c6;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

<div
  className="emotion-9 emotion-10"
  type="info"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        300
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders correctly 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e9fafe;
  border-color: #2875c6;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

<div
  className="emotion-9 emotion-10"
  type="info"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders type danger correctly 1`] = `
.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #fee3e3;
  border-color: #b93435;
}

<div
  className="emotion-9 emotion-10"
  type="danger"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders type info correctly 1`] = `
.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e9fafe;
  border-color: #2875c6;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

<div
  className="emotion-9 emotion-10"
  type="info"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders type success correctly 1`] = `
.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e0f8e6;
  border-color: #26853f;
}

<div
  className="emotion-9 emotion-10"
  type="success"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders type warning correctly 1`] = `
.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-0 {
  font-size: 0.875rem;
}

.emotion-7 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-7:focus {
  outline: none;
}

.emotion-6 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

.emotion-9 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #fdebd0;
  border-color: #e69524;
}

<div
  className="emotion-9 emotion-10"
  type="warning"
>
  <div
    className="emotion-4 emotion-5"
  >
    <div
      className="emotion-2 emotion-1"
    >
      <div
        className="emotion-0 emotion-1"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-7 emotion-8"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-6"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;

exports[`Alert renders with a title 1`] = `
.emotion-11 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  box-shadow: 0 0 0 1px rgba(0,0,0,0.06),0 3px 6px 0 rgba(0,0,0,0.03),0 1px 2px 0 rgba(0,0,0,0.1);
  border-top: 4px solid;
  border-radius: 0.375rem;
  padding: 0.75rem 1.125rem;
  background: #e9fafe;
  border-color: #2875c6;
}

.emotion-6 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.emotion-2 {
  font-size: 0.875rem;
}

.emotion-9 {
  font-size: 1rem;
  color: #0c0d0d;
  background-color: rgba(12,13,13,0.1);
  border-radius: 50%;
  border: none;
  cursor: pointer;
  width: 1em;
  height: 1em;
  padding: 0;
  position: relative;
}

.emotion-9:focus {
  outline: none;
}

.emotion-8 {
  width: 0.5em;
  height: 0.5em;
  fill: black;
  position: absolute;
  top: 0.25em;
  left: 0.25em;
  right: 0;
  bottom: 0;
}

.emotion-0 {
  margin: 0;
  color: #0c0d0d;
  font-weight: 700;
  font-size: 1rem;
  margin-top: 0;
  margin-right: 0;
  margin-bottom: 0.375rem;
  margin-left: 0;
  font-weight: 700;
}

<div
  className="emotion-11 emotion-12"
  type="info"
>
  <div
    className="emotion-6 emotion-7"
  >
    <div
      className="emotion-4 emotion-3"
    >
      <p
        className="emotion-0 emotion-1"
        color="textHeading"
        fontSize="text"
        fontWeight="bold"
      >
        My Title
      </p>
      <div
        className="emotion-2 emotion-3"
        fontSize="text-s"
      >
        Hello World
      </div>
    </div>
  </div>
  <button
    className="emotion-9 emotion-10"
    onClick={[Function]}
    type="button"
  >
    <svg
      className="emotion-8"
      viewBox="0 0 8 8"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g
        fillRule="evenodd"
        transform="translate(-4 -4)"
      >
        <path
          d="M9.414 8l2.122-2.121a1 1 0 1 0-1.415-1.415L8 6.586 5.879 4.464A1 1 0 0 0 4.464 5.88L6.586 8l-2.122 2.121a1 1 0 0 0 1.415 1.415L8 9.414l2.121 2.122a1 1 0 0 0 1.415-1.415L9.414 8z"
          fillOpacity=".5"
        />
      </g>
    </svg>
  </button>
</div>
`;
